llvm/lib/Target/AArch64/MCTargetDesc/AArch64ELFStreamer.cpp

```diff
@@ -202,6 +202,12 @@ class AArch64ELFStreamer : public MCELFStreamer {
       LastEMS = EMS_None;
 
     MCELFStreamer::changeSection(Section, Subsection);
+
+    // Section alignment of 4 to match GNU Assembler
+    if (Section->getAlign() < 4) {
```

Comment:
Yes - this should be updated to be like before @popaflorin where you were checking for if AArch64 instructions were used.

Reply:
Thank you both for your feedback! I haven't figured out how to do that yet in this AArch64ELFStreamer. I thought this will suffice: `Section->hasInstructions()` for a check but it always comes back as 0. I'm looking into it...

Reply:
it might be better/easier to check if the section is executable? that's probably the only info you have when the section is brand new (I think `hasInstructions` only becomes true when the first instruction is emitted into the section)

Reply:
I am checking now `Section->isText()` in the latest patchset which seems to get the correct executable section. In the test I have also included a `aw` section to prove that the alignment will remain unchanged at 1